src/Tizen.Multimedia.Vision/MediaVision/InferenceFaceRecognier.cs

```diff
@@ -0,0 +1,175 @@
+/*
+ * Copyright (c) 2022 Samsung Electronics Co., Ltd All Rights Reserved
+ *
+ * Licensed under the Apache License, Version 2.0 (the License);
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an AS IS BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+using System;
+using InteropFace = Interop.MediaVision.FaceRecognition;
+
+namespace Tizen.Multimedia.Vision
+{
+    /// <summary>
+    /// Provides the ability to train face for face recognition.
+    /// </summary>
+    /// <since_tizen> 10 </since_tizen>
+    public class InferenceFaceRecognizer : IDisposable
+    {
+        private IntPtr _handle;
+        private bool _disposed;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="InferenceFaceRecognizer"/> class.
+        /// </summary>
+        /// <exception cref="NotSupportedException">The required features are not supported.</exception>
+        /// <feature>http://tizen.org/feature/vision.face_recognition</feature>
+        /// <since_tizen> 10 </since_tizen>
+        public InferenceFaceRecognizer()
+        {
+            ValidationUtil.ValidateFeatureSupported(VisionFeatures.FaceRecognition);
+
+            InteropFace.Create(out _handle).Validate("Failed to create face recognizer");
+
+            try
+            {
+                InteropFace.Prepare(_handle).Validate("Failed to prepare face recognizer");
```

Comment:
Prepare method will return MediaVisionError type not throwing an exception so you may need to handle the MediaVisionError type correctly.

Reply:
I could be done with Validate() method. It's extension method for `MediaVisionError` enum.